### PR TITLE
add safety nets for type errors in amqp10 plugin

### PR DESCRIPTION
**Pull request by rochdev**
### What does this PR do?

Add safety nets for type errors in `amqp10` plugin.

### Motivation

Avoids type errors if the module API changes.

### CODE REVIEW

File: packages/datadog-plugin-amqp10/src/index.js [hunk 1]
```diff
@@ -78,22 +78,19 @@ function startReceiveSpan (tracer, config, link) {
   return span
 }
 
-function addTags (tracer, config, span, link) {
-  const address = link.session.connection.address
+function addTags (tracer, config, span, link = {}) {
+  const address = getHost(link)
```
Comment (brettlangdon) [inline packages/datadog-plugin-amqp10/src/index.js]:
```suggestion
  const address = getAddress(link)
```

?

Or maybe should adjust function or variable names?

Comment (brettlangdon) [reply packages/datadog-plugin-amqp10/src/index.js]: (or add a comment)

File: packages/datadog-plugin-amqp10/src/index.js [hunk 2]
```diff
@@ -78,22 +78,19 @@ function startReceiveSpan (tracer, config, link) {
...
 
   span.addTags({
     'service.name': config.service || `${tracer._service}-amqp`,
     'span.type': 'worker',
     'amqp.link.name': link.name,
     'amqp.link.handle': link.handle,
     'amqp.connection.host': address.host,
-    'amqp.connection.port': address.port
+    'amqp.connection.port': address.port,
```
Comment (brettlangdon) [inline packages/datadog-plugin-amqp10/src/index.js]: `null` tags still get set and sent right? do we want these to possibly show as `null` or omit them?

Comment (rochdev) [reply packages/datadog-plugin-amqp10/src/index.js]: `null` and `undefined` are skipped by the formatter.

File: packages/datadog-plugin-amqp10/src/index.js (hunk 1 above)
Comment (rochdev) [reply packages/datadog-plugin-amqp10/src/index.js]: Renamed the functions to make them clearer.

**Pull request merged**